### PR TITLE
Remove Unused env_vars From Initialization Code

### PR DESCRIPTION
**Pull request by TaekyungHeo**
## Summary
Remove unused env_vars from the initialization code.

env_vars for test templates and their strategies are initialized in the test parser class, as shown below:
```
    def _parse_data(self, data: Dict[str, Any]) -> Test:
        test_def = self.load_test_definition(data)

        env_vars = {}  # this field doesn't exist in Test or TestTemplate TOMLs
...

        test_template_name = data.get("test_template_name", "")
        test_template = self._get_test_template(test_template_name, env_vars, cmd_args)

...

        return Test(
            name=test_def.name,
            description=data.get("description", ""),
            test_template=test_template,
            env_vars=env_vars,
            cmd_args=cmd_args,
            extra_env_vars=extra_env_vars,
            extra_cmd_args=extra_cmd_args,
        )
```
As you can see, env_vars is empty by default. Therefore, there is no need to pass env_vars initially. This PR removes all env_vars arguments in the initialization logic.

## Test Plan
CI passes

### CODE REVIEW

File: src/cloudai/_core/test_template_strategy.py [hunk 1]
```diff
@@ -29,44 +29,28 @@ class TestTemplateStrategy:
...
             system (System): The system configuration for the test.
-            env_vars (Dict[str, Any]): Default environment variables.
             cmd_args (Dict[str, Any]): Default command-line arguments.
         """
         self.system = system
         self.install_path = ""
-        self.env_vars = env_vars
         self.cmd_args = cmd_args
-        self.default_env_vars = self._construct_default_env_vars()
+        self.default_env_vars = {}
```
Comment (amaslenn) [inline src/cloudai/_core/test_template_strategy.py:50]: Quick search shows that we only update `self.default_env_vars`, but never actually use it. Could you please double check and remove if it is not used?

Comment (TaekyungHeo) [reply src/cloudai/_core/test_template_strategy.py:50]: We use `self.default_env_vars`. The `default_env_vars` are used to initialize or update `final_env_vars`, which is then utilized for command generation. We cannot remove `default_env_vars`. I will rename and refactor `default_env_vars` and `default_cmd_args` based on our discussion in follow-up PRs.

Comment (TaekyungHeo) [reply src/cloudai/_core/test_template_strategy.py:50]:
```
src/cloudai/_core/test_template_strategy.py:        default_env_vars (Dict[str, str]): Constructed default environment variables.
src/cloudai/_core/test_template_strategy.py:        self.default_env_vars = {}
src/cloudai/_core/test_template_strategy.py:        self.default_env_vars.update(system.global_env_vars)
src/cloudai/_core/test_template_strategy.py:        default_env_vars: Dict[str, str],
src/cloudai/_core/test_template_strategy.py:            default_env_vars (Dict[str, str]): The default environment variables.
src/cloudai/_core/test_template_strategy.py:        final_env_vars = default_env_vars.copy()
src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/sleep/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/ucc_test/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nemo_launcher/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/chakra_replay/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nccl_test/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nccl_test/kubernetes_json_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
```

Comment (TaekyungHeo) [reply src/cloudai/_core/test_template_strategy.py]: Got it, updated.

**Review by amaslenn — DISMISSED** [review]

**Comment by TaekyungHeo** [comment]
Created a merge commit. Please re-approve, @amaslenn .

**Review by amaslenn — APPROVED** [review]

**Pull request merged**